node/src/components/linear_chain_fast_sync.rs

```diff
@@ -131,28 +138,22 @@ impl<I: Clone + PartialEq + 'static> LinearChainFastSync<I> {
         let mut curr_state = mem::replace(&mut self.state, State::None);
         match curr_state {
             State::None | State::Done => panic!("Block handled when in {:?} state.", &curr_state),
-            // Keep syncing from genesis if we haven't reached the trusted block hash
             State::SyncingTrustedHash {
                 highest_block_seen,
-                ref mut validator_weights,
+                trusted_header: None,
                 ..
-            } if highest_block_seen != block_height => {
-                if let Some(validator_weights_for_new_era) =
-                    block_header.next_era_validator_weights()
-                {
-                    *validator_weights = validator_weights_for_new_era.clone();
-                }
-                self.state = curr_state;
-                self.fetch_next_block_deploys(effect_builder)
-            }
-            // Otherwise transition to State::SyncingDescendants
+            } if highest_block_seen == block_height => panic!("Should always have trusted header"),
+            // If the block we are handling is the highest block seen, transition to syncing
+            // descendants
             State::SyncingTrustedHash {
                 highest_block_seen,
                 trusted_hash,
+                trusted_header,
                 ref latest_block,
                 validator_weights,
                 ..
-            } => {
+            } if highest_block_seen == block_height => {
+                // TODO: Fail gracefully in these cases
                 assert_eq!(highest_block_seen, block_height);
```

Comment:
That assertion is redundant now, since it's in the `match` condition itself.

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/806/commits/240258f15750eccd6ad0a95086463f1d75674af8